### PR TITLE
GM Changes for JS SDK

### PR DESCRIPTION
**Pull request by swarajban**
Basic functionality is done. I now need to make sure VN & starter app can specify domain ID & add missing JSDoc.

### CODE REVIEW

File: src/sdk/main.js [hunk 1]
```diff
@@ -3458,8 +3417,42 @@ var MM = ( function (window, ajax, Faye) {
...
+                var deleteDocumentPath = 'domain/' + this.domainID + '/document/' + documentID;
+                this.makeModelRequest('DELETE', deleteDocumentPath, null, onSuccess, onFail);
+            } else {
+                MM.Util.testAndCall(onFail, MM.Internal.noDomainError);
+            }
+
+        }
+    });
+
+    MM.models.AppDomainList = new MM.Internal.createSubclass(MM.models.Model, {
```
Comment (swarajban) [inline src/sdk/main.js]: Need to document this

File: src/test/spec/data/fakeResponses.js [hunk 2]
```diff
@@ -340,6 +373,6 @@ window.FakeResponses = {
         }
       }
     };
-  },
```
Comment (swarajban) [inline src/test/spec/data/fakeResponses.js]: @jagill intellij complains about trailing commas, sorry atom :)

Comment (jagill) [inline src/test/spec/data/fakeResponses.js]: Change your linter settings! :)

File: src/sdk/main.js [hunk 3]
```diff
@@ -819,8 +842,7 @@ var MM = ( function (window, ajax, Faye) {
             onError('You must supply the appid in the config object.');
           }
 
-          var makeAnonymousCredentials = function () {
-
+          function makeAnonymousCredentials () {
```
Comment (jagill) [inline src/sdk/main.js]: Code style!  We prefer 'var foo = function () {}`.

File: src/sdk/main.js [hunk 4]
```diff
@@ -838,12 +860,12 @@ var MM = ( function (window, ajax, Faye) {
                 domain: window.location.hostname
               }
             };
-          };
+          }
 
-          var handleSession = function () {
+          function handleSession () {
```
Comment (jagill) [inline src/sdk/main.js]: Code style!  We still prefer 'var foo = function () {}`.

File: src/sdk/main.js [hunk 5]
```diff
@@ -859,12 +881,32 @@ var MM = ( function (window, ajax, Faye) {
...
+                  MM.setActiveSessionID(result.data.sessionid, handleDomain, onError);
                 },
                 onError
               );
             }
-          };
+          }
+
+          // Sets MM.activeDomainID either from config, or by fetching domains
+          function handleDomain () {
```
Comment (jagill) [inline src/sdk/main.js]: Code style, part III.

File: src/sdk/main.js [hunk 6]
```diff
@@ -3282,15 +3228,16 @@ var MM = ( function (window, ajax, Faye) {
...
         localStoragePath: function () {
             return 'MM.documents';
         },
         path: function () {
-            return('documents');
+            return 'domain/' + this.domainID + '/documents';
         },
         /**
-         * Helper function returns the JSON data from the application's document collection
+         * Helper function returns thMe JSON data from the application's document collection
```
Comment (jagill) [inline src/sdk/main.js]: Typo.

File: example/sdk/multiLingualVoice/js/app.js [hunk 7]
```diff
@@ -91,8 +91,9 @@
         ['Lingua latīna',         ['la']]
     ];
 
-    var MM_APP_ID = 'ENTER_YOUR_APP_ID';
-    var MM_APP_SECRET = 'ENTER_YOUR_APP_SECRET';
+    var MM_APP_ID = 'a2b258bde95545580b8ec16994642bc74afe8d31';
+    var MM_APP_SECRET = 'fd90ccb6b5bf71a640ccba2bd66ff5018607d032';
+    var MM_DOMAIN_ID = '1989';
```
Comment (jjackson26) [inline example/sdk/multiLingualVoice/js/app.js]: Do we want these constants checked in?

File: src/sdk/main.js (hunk 3 above)
Comment (jjackson26) [inline src/sdk/main.js]: +1

File: src/sdk/main.js [hunk 8]
```diff
@@ -3458,8 +3417,42 @@ var MM = ( function (window, ajax, Faye) {
...
+        path: function () {
+            return 'domains';
+        },
+        json: function () {
+            return this._json();
+        },
+        get: function (onSuccess, onFail) {
+            this._get(null, onSuccess, onFail);
+        },
+        createNew: function (name, onSuccess, onFail) {
```
Comment (jjackson26) [inline src/sdk/main.js]: Why `createNew` and not `post`?

File: src/sdk/main.js [hunk 9]
```diff
@@ -3458,8 +3417,42 @@ var MM = ( function (window, ajax, Faye) {
             // document with documentid response.data.documentid deleted
          }
          */
-        delete: function (documentid, onSuccess, onFail) {
-            this.makeModelRequest('DELETE', 'document/' + documentid, null, onSuccess, onFail);
+        delete: function (documentID, onSuccess, onFail, domainID) {
+            this.domainID = MM.Internal.getDomainID(domainID);
+            if (this.domainID !== null) {
+                var deleteDocumentPath = 'domain/' + this.domainID + '/document/' + documentID;
```
Comment (jjackson26) [inline src/sdk/main.js]: Doesn't the mini GM mean we can use `DELETE /document/:documentid`? Maybe it's not implemented yet.

Comment (swarajban) [inline src/sdk/main.js]: It does mean that, but we have not yet implemented this API endpoint. I did not want to block this release for API updates

File: src/sdk/main.js (hunk 8 above)
Comment (swarajban) [inline src/sdk/main.js]: Though it's less consistent with other objects, I think it's a better name in the SDK. I think we should make more user-friendly function names like this for other methods. I think making small changes on the way is a good way towards that. 

Do you still think it should be `post`?

File: example/sdk/multiLingualVoice/js/app.js (hunk 7 above)
Comment (swarajban) [inline example/sdk/multiLingualVoice/js/app.js]: Oops

File: src/sdk/main.js (hunk 3 above)
Comment (swarajban) [inline src/sdk/main.js]: Argh, I actually like this way, but guys are right!

**Pull request merged**